syntax = "proto3";
package qkd;

message OpenConnectRequest {
    string client_id = 1;
    string auth_token = 2;
}

message OpenConnectResponse {
    bool success = 1;
    string message = 2;
    string session_token = 3;
}

message GetKeyRequest {
    string key_id = 1;
    string session_token = 2;
}

message GetKeyResponse {
    string key_data = 1; // base64 encoded
    bool success = 2;
    string message = 3;
    string session_token = 4;
}

message CloseRequest {
    string client_id = 1;
    string session_token = 2;
}

message CloseResponse {
    bool success = 1;
    string message = 2;
}

service QKDKeyManagementService {
    rpc OpenConnect(OpenConnectRequest) returns (OpenConnectResponse);
    rpc GetKey(GetKeyRequest) returns (GetKeyResponse);
    rpc Close(CloseRequest) returns (CloseResponse);
}
